src/buffer/out/textBuffer.cpp

```diff
@@ -3185,23 +3185,23 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
 
 // Searches through the entire (committed) text buffer for `needle` and returns the coordinates in absolute coordinates.
 // The end coordinates of the returned ranges are considered inclusive.
-std::vector<til::point_span> TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags) const
+bool TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags, std::vector<til::point_span>& results) const
 {
-    return SearchText(needle, flags, 0, til::CoordTypeMax);
+    return SearchText(needle, flags, 0, til::CoordTypeMax, results);
 }
 
 // Searches through the given rows [rowBeg,rowEnd) for `needle` and returns the coordinates in absolute coordinates.
 // While the end coordinates of the returned ranges are considered inclusive, the [rowBeg,rowEnd) range is half-open.
-std::vector<til::point_span> TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags, til::CoordType rowBeg, til::CoordType rowEnd) const
+bool TextBuffer::SearchText(const std::wstring_view& needle, SearchFlag flags, til::CoordType rowBeg, til::CoordType rowEnd, std::vector<til::point_span>& results) const

```

Comment:
The method description could use a minor update. I'd particularly like an added description for what it means for this function to return `false`. Is it only if the provided regex was invalid?